panther_hardware_interfaces/include/panther_hardware_interfaces/roboteq_data_converters.hpp

```diff
@@ -0,0 +1,277 @@
+#define PANTHER_HARDWARE_INTERFACES_ROBOTEQ_DATA_CONVERTERS_HPP_
+
+#include <vector>
+
+#include <panther_msgs/msg/fault_flag.hpp>
+#include <panther_msgs/msg/runtime_error.hpp>
+#include <panther_msgs/msg/script_flag.hpp>
+
+#include <panther_hardware_interfaces/roboteq_driver.hpp>
+#include <panther_hardware_interfaces/utils.hpp>
+
+namespace panther_hardware_interfaces
+{
+struct DrivetrainSettings
+{
+  float motor_torque_constant;
+  float gear_ratio;
+  float gearbox_efficiency;
+  float encoder_resolution;
+  float max_rpm_motor_speed;
+};
+
+/**
+ * @brief Class for converting velocity commands (between SI units and raw Roboteq cmd)
+ */
+class RoboteqVeloctiyCommandConverter
+{
+public:
+  RoboteqVeloctiyCommandConverter(DrivetrainSettings drivetrain_settings);
+  int32_t Convert(float cmd) const { return LimitCmd(cmd * radians_per_second_to_roboteq_cmd_); }
+
+private:
+  inline int32_t LimitCmd(int32_t cmd) const
+  {
+    return std::clamp(cmd, -max_roboteq_cmd_value_, max_roboteq_cmd_value_);
+  }
+
+  float radians_per_second_to_roboteq_cmd_;
+  static constexpr int32_t max_roboteq_cmd_value_ = 1000;
+};
+
+/**
+ * @brief Class for storing and converting (between raw Roboteq data and SI units) motor
+ * state (position, velocity, torque)
+ */
+class MotorState
+{
+public:
+  MotorState(DrivetrainSettings drivetrain_settings);
+
+  void SetData(RoboteqMotorState fb) { last_state_ = fb; };
+
+  float GetPosition() const { return last_state_.pos * roboteq_pos_feedback_to_radians_; }
+  float GetVelocity() const
+  {
+    return last_state_.vel * roboteq_vel_feedback_to_radians_per_second_;
+  }
+  float GetTorque() const
+  {
+    return last_state_.current * roboteq_current_feedback_to_newton_meters_;
+  }
+
+private:
+  float roboteq_pos_feedback_to_radians_;
+  float roboteq_vel_feedback_to_radians_per_second_;
+  float roboteq_current_feedback_to_newton_meters_;
+
+  RoboteqMotorState last_state_ = {0, 0, 0};
+};
+
+/**
+ * @brief Base class for different errors that are based on setting bits of one byte
+ */
+class FlagError
+{
+public:
+  /**
+   * @brief Sets which flags should be ignored when checking if an error occurred when converting
+   * to message true data still will be set
+   */
+  FlagError(
+    const std::vector<std::string> & flag_names,
+    const std::vector<std::string> & surpressed_flags_names = {})
+  : flag_names_(flag_names)
+  {
+    for (size_t i = 0; i < surpressed_flags_names.size(); ++i) {
+      for (size_t j = 0; j < flag_names_.size(); ++j) {
+        if (surpressed_flags_names[i] == flag_names_[j]) {
+          surpressed_flags_ = SetBit(surpressed_flags_, j);
+        }
+      }
+    }
+  }
+
+  void SetData(uint8_t flags) { flags_ = flags; }
+
+  bool IsError() const { return (flags_ & (~surpressed_flags_)) != 0; }
+
+  std::string GetErrorLog() const;
+
+protected:
+  const std::vector<std::string> flag_names_;
+
+  uint8_t surpressed_flags_ = 0b00000000;
+  uint8_t flags_ = 0b00000000;
```

Comment:
```suggestion
  uint8_t surpressed_flags_ = 0x00;
  uint8_t flags_ = 0x00;
```
I think using hex is as easy to understand as bits

Reply:
maybe i'll just change it to 0, as it doesn't really make sense to use bin or hex here